Composer/packages/client/jest.config.js

```diff
@@ -10,6 +10,8 @@ module.exports = {
     // Any imports of .scss / .css files will instead import styleMock.js which is an empty object
     '\\.(jpg|jpeg|png|svg|gif)$': '<rootDir>/__tests__/jestMocks/styleMock.js',
     '\\.(s)?css$': '<rootDir>/__tests__/jestMocks/styleMock.js',
+    // lsp code editor
+    vscode: 'monaco-languageclient/lib/vscode-compatibility',
```

Comment:
I don't understand why this is needed. I think jest will only map to this module if any source code of client imports `vscode`. It doesn't look like it does.

Reply:
Installed `vscode` package only contain type declaration, ( last commit replaced with `@types/vscode` ), so we can't import it.
The dependency chain is like this:
`LgEditor` --> `monaco-languageclient`  -->  `vscode-languageclient`  --> `vscode`

`vscode-languageclient`  run at vscode-extension environment (desktop application).
`monaco-languageclient` run at web browser, so it mock a vscode environment, which is 'monaco-languageclient/lib/vscode-compatibility'. it need to be manually import in webpack.config.js 's alias, and here for jest.

Reply:
Ok. I'll investigate this today.

Reply:
fixed, awesome!